src/web/lib/useSignInGateSelector.ts

```diff
@@ -0,0 +1,42 @@
+import { Runnable } from '@guardian/ab-core';
+import { useAB } from '@guardian/ab-react';
+import {
+	SignInGateComponent,
+	CurrentSignInGateABTest,
+} from '@frontend/web/components/SignInGate/types';
+import {
+	signInGateTests,
+	signInGateTestVariantToGateMapping,
+} from '@frontend/web/components/SignInGate/signInGate';
+
+/**
+ * @description
+ * A custom hook to make which selects the sign in gate (component and ab test)
+ * to be displayed on the current page
+ * */
+export const useSignInGateSelector = ():
+	| [SignInGateComponent, CurrentSignInGateABTest]
+	| null => {
+	const ab = useAB();
```

Comment:
this used to get executed with useEffect with dependency `[ab]` which then updated some state. 

Does `ab` never change?